gnucash/gnome/gnc-split-reg.c

```diff
@@ -402,6 +407,57 @@ gsr_setup_table( GNCSplitReg *gsr )
     LEAVE(" ");
 }
 
+
+static gchar *
+gnc_split_reg_get_state_group (SplitRegister *split_reg)
+{
+    switch (split_reg->type)
+    {
+        case BANK_REGISTER:
+        case CASH_REGISTER:
+        case ASSET_REGISTER:
+        case CREDIT_REGISTER:
+        case LIABILITY_REGISTER:
+        case INCOME_REGISTER:
+        case EXPENSE_REGISTER:
+        case EQUITY_REGISTER:
+        case TRADING_REGISTER:
+        {
+            return g_strdup ("Register Default State Group1");
+            break;
+        }
+        case PAYABLE_REGISTER:
+        case RECEIVABLE_REGISTER:
+        {
+            return g_strdup ("Register Default State Group2");
+            break;
+        }
+        case INCOME_LEDGER:
+        case GENERAL_JOURNAL:
+        case SEARCH_LEDGER:
+        {
+            return g_strdup ("Register Default State Group3");
+            break;
+        }
+        case STOCK_REGISTER:
+        case CURRENCY_REGISTER:
+        {
+            return g_strdup ("Register Default State Group4");
+            break;
+        }
+        case PORTFOLIO_LEDGER:
+        {
+            return g_strdup ("Register Default State Group5");
+            break;
+        }
```

Comment:
What is this grouping of register types based on ?

Reply:
I looked at gnc_split_register_set_cells and used the same switch statement.

Reply:
Ok, that seems like a good start. However just copying this over is fairly fragile. Clearly both code locations want to group account types for the same purpose. It may be better to factor this grouping out into a separate function and have it return a unique group identifier. I'm thinking of a separate enum named "SplitRegisterTypeGroup" with 5 values; one for each of the groups you define above.

I think these enum names may be fitting:
REG_TYPE_GROUP_SIMPLE (or REG_TYPE_GROUP_CURRENCY)
REG_TYPE_GROUP_APAR
REG_TYPE_GROUP_STOCK
REG_TYPE_GROUP_JOURNAL
REG_TYPE_GROUP_PORTFOLIO

With that you can also define a list of (translatable) names for these.
As a first suggestion:
- Currency account registers
- Business account registers
- Stock account registers
- Journal registers
- Portfolio registers

Reply:
OK, Have almost got this sorted, just wondering if I need both translated for the menu and an untranslated for the state group section. Not sure on the policy of having translated group sections in the state file?

Reply:
I'm sorry for my delayed response. I think the state file should use untranslated section names. Otherwise just changing language would cause gnucash to no longer recognize the sections.

Reply:
(Oh I did reply but via mail... apparently that fails to add the reply to the proper thread)